lineage/states/stateCommon.py

```diff
@@ -135,11 +135,16 @@ def gamma_estimator_atonce(gamma_obs, time_cen, gamas, x0=None):
     if x0 is None:
         x0 = np.array([20.0, 2.0, 3.0, 4.0, 5.0])
 
-    def constr(x):
-        return x[2:5] - x[1:4]
+    A = np.zeros((3, 5))
+    np.fill_diagonal(A[:, 1:], -1.0)
+    np.fill_diagonal(A[:, 2:], 1.0)
 
-    def constr2(x):
-        return 100.0 - x
+    # Override x0 if we were given a bad starting point
+    if np.allclose(np.dot(A, x0), 0.0):
```

Comment:
We might need to also check `x0` before starting, and if it's violating the bounds move it back to a reasonable start.

Reply:
Thank you.
This is the reason I changed it to `cobyla` but you are right, it should be an easy fix.